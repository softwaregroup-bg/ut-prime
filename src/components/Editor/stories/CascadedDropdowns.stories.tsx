import { userEvent, within } from '@storybook/testing-library';
import React from 'react';
import Editor from '..';
export {default} from '../Editor.stories';

export const CascadedDropdowns = () =>
    <Editor
        schema={{
            type: 'object',
            properties: {
                continent: {widget: {type: 'dropdown', dropdown: 'continent'}},
                country: {widget: { type: 'dropdown', dropdown: 'country', parent: 'continent'}},
                city: {widget: { type: 'dropdown', dropdown: 'city', parent: 'country'}}
            }
        }}
        cards={{
            edit: {
                className: 'xl:col-3',
                widgets: ['continent', 'country', 'city']
            }
        }}
        onDropdown={() => Promise.resolve({
            continent: [
                {value: 1, label: 'Africa'},
                {value: 2, label: 'Asia'},
                {value: 3, label: 'Europe'},
                {value: 4, label: 'North America'},
                {value: 5, label: 'Oceania'},
                {value: 6, label: 'South America'}
            ],
            country: [
                {value: 1, label: 'Kenya', parent: 1},
                {value: 2, label: 'Uganda', parent: 1},
                {value: 3, label: 'Tanzania', parent: 1},
                {value: 7, label: 'United States', parent: 4},
                {value: 9, label: 'Mexico', parent: 4},
                {value: 10, label: 'Australia', parent: 5},
                {value: 11, label: 'New Zealand', parent: 5},
                {value: 12, label: 'France', parent: 3},
                {value: 13, label: 'Germany', parent: 3},
                {value: 14, label: 'Italy', parent: 3},
                {value: 15, label: 'Spain', parent: 3}
            ],
            city: [
                {value: 1, label: 'Nairobi', parent: 1},
                {value: 2, label: 'Kampala', parent: 2},
                {value: 3, label: 'Dar es Salaam', parent: 3},
                {value: 4, label: 'New York', parent: 7},
                {value: 6, label: 'Mexico City', parent: 9},
                {value: 7, label: 'Canberra', parent: 10},
                {value: 8, label: 'Paris', parent: 12},
                {value: 9, label: 'Berlin', parent: 13},
                {value: 10, label: 'Rome', parent: 14}
            ]
        })}
        layouts={{edit: ['edit']}}
        toolbar
    />;

CascadedDropdowns.play = async({canvasElement}) => {
    const canvas = within(canvasElement);
    const body = within(document.body);
    userEvent.click(await canvas.findByTestId('continent'));
    userEvent.click(body.getByText('Europe'));
    userEvent.click(canvas.getByTestId('country'));
    userEvent.click(body.getByText('France'));
    userEvent.click(canvas.getByTestId('city'));
    userEvent.click(body.getByText('Paris'));
};
