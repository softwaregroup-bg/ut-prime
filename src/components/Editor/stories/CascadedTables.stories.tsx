import { userEvent, within } from '@storybook/testing-library';
import React from 'react';
import Editor from '..';
export {default} from '../Editor.stories';

export const CascadedTables = () =>
    <Editor
        id = {1}
        onGet={() => Promise.resolve({
            person: [
                {personId: 1, fullName: 'John Doe', birthDate: '2000-01-08', nationalId: '777777777'},
                {personId: 2, fullName: 'Jane Doe', birthDate: '2002-03-18', nationalId: '888888888'}
            ],
            document: [
                {documentId: 1, personId: 1, documentTypeId: 1, documentType: 'Passport'},
                {documentId: 2, personId: 1, documentTypeId: 2, documentType: 'Driving License'},
                {documentId: 3, personId: 1, documentTypeId: 3, documentType: 'Marriage Certificate'},
                {documentId: 4, personId: 2, documentTypeId: 1, documentType: 'Passport'},
                {documentId: 5, personId: 2, documentTypeId: 2, documentType: 'Driving License'}

            ].map(item => ({...item, issueDate: '2020-01-0' + item.documentId, expiryDate: '2025-01-0' + item.documentId})),
            attachment: [
                {attachmentId: 1, documentId: 1, pageNumber: 1, contentType: 'image/jpeg', sizeBytes: 12345},
                {attachmentId: 1, documentId: 1, pageNumber: 2, contentType: 'image/jpeg', sizeBytes: 23456},
                {attachmentId: 1, documentId: 2, pageNumber: 1, contentType: 'image/png', sizeBytes: 33333},
                {attachmentId: 1, documentId: 2, pageNumber: 2, contentType: 'image/png', sizeBytes: 22222}
            ]
        })}
        onDropdown={names => Promise.resolve({
            documentType: [
                {value: 1, label: 'Passport'},
                {value: 2, label: 'Driving License'},
                {value: 3, label: 'Marriage Certificate'}
            ]
        })}
        cards={{
            person: {
                className: 'xl:col-4',
                label: 'Person',
                widgets: [{
                    name: 'person',
                    hidden: ['personId'],
                    widgets: ['fullName', 'nationalId', 'birthDate']
                }]
            },
            document: {
                className: 'xl:col-4',
                label: 'Document',
                widgets: [{
                    name: 'document',
                    hidden: ['documentId', 'personId'],
                    widgets: ['documentTypeId', 'issueDate', 'expiryDate']
                }]
            },
            attachment: {
                className: 'xl:col-4',
                label: 'Attachment',
                widgets: [{
                    name: 'attachment',
                    hidden: ['attachmentId', 'documentId'],
                    widgets: ['pageNumber', 'contentType', 'sizeBytes']
                }]
            }
        }}
        schema={{
            type: 'object',
            properties: {
                person: {
                    type: 'array',
                    items: {
                        type: 'object',
                        properties: {
                            fullName: {},
                            nationalId: {title: 'National ID'},
                            birthDate: {format: 'date'}
                        }
                    },
                    title: '',
                    widget: {
                        type: 'table',
                        selectionMode: 'single'
                    }
                },
                document: {
                    type: 'array',
                    items: {
                        properties: {
                            documentTypeId: {body: 'documentType', widget: {type: 'dropdown', dropdown: 'documentType'}},
                            issueDate: {format: 'date'},
                            expiryDate: {format: 'date'}
                        }
                    },
                    title: '',
                    widget: {
                        type: 'table',
                        master: {
                            personId: 'personId'
                        },
                        parent: '$.selected.person',
                        autoSelect: true,
                        selectionMode: 'single'
                    }
                },
                attachment: {
                    type: 'array',
                    items: {
                        properties: {
                            pageNumber: {type: 'integer'},
                            contentType: {},
                            sizeBytes: {type: 'integer', title: 'Size (bytes)'}
                        }
                    },
                    title: '',
                    widget: {
                        type: 'table',
                        master: {
                            documentId: 'documentId'
                        },
                        parent: '$.selected.document',
                        autoSelect: true,
                        selectionMode: 'single'
                    }
                }
            }
        }}
        layouts={{edit: ['person', 'document', 'attachment']}}
    />;

CascadedTables.play = async({canvasElement}) => {
    const canvas = within(canvasElement);
    userEvent.click(await canvas.findByText('John Doe'));
    userEvent.click(canvas.getByText('Driving License'));
};
