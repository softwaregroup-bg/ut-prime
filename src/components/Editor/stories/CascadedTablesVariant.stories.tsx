import React from 'react';
import type { Story } from '@storybook/react';
import type { Props } from '../Editor.types';
import Editor from '..';
import useToast from '../../hooks/useToast';

export { default } from '../Editor.stories';

const Template: Story<Props> = ({...args}) => {
    const {toast, submit} = useToast();
    return <>
        <Editor
            id={1}
            onGet={() =>
                Promise.resolve({
                    roleCategory: [
                        {actionCategoryId: 478, label: 'Manage Users', hasSettings: true},
                        {actionCategoryId: 681, label: 'Manage History', hasSettings: false},
                        {actionCategoryId: 259, label: 'Manage Content Items and Translations', hasSettings: false}
                    ],
                    permission: [
                        {actionCategoryId: 259, actionId: 1006, actionName: 'Add Item', hasRight: true, isOwn: false, objectIds: null },
                        {actionCategoryId: 259, actionId: 1025, actionName: 'Edit Item', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 259, actionId: 1029, actionName: 'Enable / Disable Item', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 259, actionId: 1032, actionName: 'List Items', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 259, actionId: 1047, actionName: 'Upload Items', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 259, actionId: 1052, actionName: 'View Item', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 478, actionId: 1008, actionName: 'Add User', hasRight: true, isOwn: true, objectIds: [1167, 1168, 1169]},
                        {actionCategoryId: 478, actionId: 1011, actionName: 'Authorize Changes (Approve/Reject)', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 478, actionId: 1014, actionName: 'Clear Unsuccessful Login Attempts', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 478, actionId: 1020, actionName: 'Delete User', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 478, actionId: 1028, actionName: 'Edit User', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 478, actionId: 1101, actionName: 'List Users', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 478, actionId: 1035, actionName: 'Lock / Unlock User', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 478, actionId: 1055, actionName: 'View User', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 681, actionId: 1109, actionName: 'Access Policy History Log', hasRight: true, isOwn: false, objectIds: [1173, 1171]},
                        {actionCategoryId: 681, actionId: 1074, actionName: 'Agents and Merchants Network History Log', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 681, actionId: 1144, actionName: 'Business Unit History Log', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 681, actionId: 1150, actionName: 'Export History', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 681, actionId: 1092, actionName: 'KYC History Log', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 681, actionId: 1065, actionName: 'Role History Log', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 681, actionId: 1136, actionName: 'Rule History Log', hasRight: false, isOwn: false, objectIds: null},
                        {actionCategoryId: 681, actionId: 1066, actionName: 'User History Log', hasRight: true, isOwn: false, objectIds: null}
                    ],
                    permissionRole: [
                        { actionId: 1008, value: 1167, selected: true, label: 'Banks Branch User', description: 'Banks Branch User' },
                        { actionId: 1008, value: 1168, selected: true, label: 'CBI Admin', description: 'CBI Admin' },
                        { actionId: 1008, value: 1169, selected: true, label: 'CBI Auditor', description: 'CBI Auditor' },
                        { actionId: 1008, value: 1170, selected: false, label: 'CBI Compliance Office User', description: 'CBI Compliance Office User' },
                        { actionId: 1066, value: 1171, selected: true, label: 'CBI Directorate Admin Checker', description: 'CBI Directorate Admin Checker' },
                        { actionId: 1066, value: 1172, selected: false, label: 'CBI Directorate Admin Maker', description: 'CBI Directorate Admin Maker' },
                        { actionId: 1066, value: 1173, selected: true, label: 'CBI IT Admin Checker', description: 'CBI IT Admin Checker' },
                        { actionId: 1009, value: 1164, selected: false, label: 'Bank Admin Checker', description: 'Bank Admin Checker' },
                        { actionId: 1109, value: 1165, selected: false, label: 'Bank Admin Maker', description: 'Bank Admin Maker' },
                        { actionId: 1109, value: 1166, selected: false, label: 'Bank User', description: 'Bank User' }
                    ],
                    document: [
                        {documentId: 1, documentTypeId: 1, documentType: 'Passport', check: true},
                        {documentId: 2, documentTypeId: 2, documentType: 'Driving License', check: false},
                        {documentId: 3, documentTypeId: 3, documentType: 'Marriage Certificate', check: true},
                        {documentId: 4, documentTypeId: 1, documentType: 'Passport', check: false},
                        {documentId: 5, personId: 2, documentTypeId: 2, documentType: 'Driving License', check: true}
                    ].map(item => ({...item, issueDate: '2020-01-0' + item.documentId, expiryDate: '2025-01-0' + item.documentId}))
                })
            }
            onDropdown={() =>
                Promise.resolve({
                    permissionRole: [
                        { value: 1164, label: 'Bank Admin Checker', description: 'Bank Admin Checker' },
                        { value: 1165, label: 'Bank Admin Maker', description: 'Bank Admin Maker' },
                        { value: 1166, label: 'Bank User', description: 'Bank User' },
                        { value: 1167, label: 'Banks Branch User', description: 'Banks Branch User' },
                        { value: 1168, label: 'CBI Admin', description: 'CBI Admin' },
                        { value: 1169, label: 'CBI Auditor', description: 'CBI Auditor' },
                        { value: 1170, label: 'CBI Compliance Office User', description: 'CBI Compliance Office User' },
                        { value: 1171, label: 'CBI Directorate Admin Checker', description: 'CBI Directorate Admin Checker' },
                        { value: 1172, label: 'CBI Directorate Admin Maker', description: 'CBI Directorate Admin Maker' },
                        { value: 1173, label: 'CBI IT Admin Checker', description: 'CBI IT Admin Checker' },
                        { value: 1174, label: 'CBI IT Admin Maker', description: 'CBI IT Admin Maker' },
                        { value: 1175, label: 'CBI System User', description: 'CBI System User' },
                        { value: 1176, label: 'CSW Admin Maker', description: 'CSW Admin Maker' },
                        { value: 1177, label: 'EXC Admin Checker', description: 'EXC Admin Checker' },
                        { value: 1178, label: 'EXC Admin Maker', description: 'EXC Admin Maker' },
                        { value: 1179, label: 'EXC Branch User', description: 'EXC Branch User' },
                        { value: 1180, label: 'EXC User', description: 'EXC User' }
                    ],
                    documentType: [
                        {value: 1, label: 'Passport'},
                        {value: 2, label: 'Driving License'},
                        {value: 3, label: 'Marriage Certificate'}
                    ]
                })
            }
            onEdit={submit}
            {...args}
        />
        {toast}
    </>;
};

declare type StoryTemplate = Story<Partial<Props>> & {
    play: (context: {canvasElement: HTMLElement}) => Promise<void>
}

export const CascadedTablesVariant: StoryTemplate = Template.bind({});
CascadedTablesVariant.args = {
    schema: {
        type: 'object',
        properties: {
            roleCategory: {
                type: 'array',
                title: '',
                items: {
                    type: 'object',
                    readOnly: true,
                    properties: {
                        actionCategoryId: {},
                        label: {
                            title: 'Category',
                            readOnly: true
                        }
                    }
                },
                widget: {
                    type: 'table',
                    selectionMode: 'single',
                    actions: {
                        allowAdd: false,
                        allowDelete: false,
                        allowEdit: false
                    }
                }
            },
            permission: {
                title: '',
                items: {
                    type: 'object',
                    properties: {
                        actionCategoryId: {},
                        actionId: {},
                        actionName: {
                            title: 'Granted Permissions',
                            readOnly: true
                        },
                        hasRight: {
                            title: ' ',
                            widget: {
                                column: {
                                    className: 'w-1'
                                },
                                type: 'boolean',
                                inlineEdit: true
                            }
                        },
                        isOwn: {
                            widget: {
                                type: 'boolean'
                            }
                        },
                        objectIds: {}
                    }
                },
                widget: {
                    title: 'Granted Permissions',
                    type: 'table',
                    selectionMode: 'single',
                    actions: {
                        allowAdd: false,
                        allowDelete: false,
                        allowEdit: false
                    },
                    hidden: ['actionCategoryId', 'actionId', 'isOwn', 'objectIds'],
                    widgets: ['actionName', 'hasRight'],
                    parent: '$.selected.roleCategory',
                    master: {
                        actionCategoryId: 'actionCategoryId'
                    }
                }
            },
            permissionRole: {
                title: '',
                items: {
                    type: 'object',
                    properties: {
                        actionId: {},
                        selected: {
                            title: ' ',
                            widget: {
                                column: {
                                    className: 'w-1'
                                },
                                type: 'boolean',
                                inlineEdit: true
                            }
                        },
                        value: {},
                        label: {
                            title: 'Permissions for Role'
                        }
                    }
                },
                widget: {
                    type: 'table',
                    selectionMode: 'single',
                    actions: {
                        allowAdd: false,
                        allowDelete: false,
                        allowEdit: false
                    },
                    hidden: ['actionId', 'value'],
                    widgets: ['selected', 'label'],
                    parent: '$.selected.permission',
                    master: {
                        actionId: 'actionId'
                    }
                }
            },
            document: {
                type: 'array',
                items: {
                    properties: {
                        documentTypeId: {
                            body: 'documentType',
                            widget: {
                                type: 'dropdown',
                                dropdown: 'documentType',
                                inlineEdit: true,
                                showClear: false
                            }
                        },
                        check: {
                            widget: {
                                column: {
                                    className: 'w-1'
                                },
                                type: 'boolean',
                                inlineEdit: true
                            }
                        },
                        issueDate: {format: 'date'},
                        expiryDate: {
                            format: 'date',
                            widget: {
                                inlineEdit: true,
                                showClear: false
                            }
                        }
                    }
                },
                title: '',
                widget: {
                    actions: {
                        allowEdit: false
                    },
                    type: 'table',
                    autoSelect: true,
                    selectionMode: 'single'
                }
            }
        }
    },
    cards: {
        roleCategory: {
            className: 'md:col-8 lg:col-4 xl:col-3',
            widgets: [{
                name: 'roleCategory',
                hidden: ['actionCategoryId'],
                widgets: ['label']
            }]
        },
        permission: {
            className: 'md:col-8 lg:col-4 xl:col-3',
            widgets: [{
                name: 'permission',
                hidden: ['actionCategoryId', 'actionId', 'actionName', 'hasRight', 'isOwn'],
                widgets: ['hasRight', 'actionName']
            }]
        },
        permissionRole: {
            className: 'md:col-8 lg:col-4 xl:col-3',
            enabled: '$.selected.roleCategory.hasSettings',
            widgets: [{
                name: 'permissionRole',
                hidden: ['actionId', 'value'],
                widgets: ['selected', 'label']
            }]
        },
        document: {
            label: 'Document',
            widgets: [{
                name: 'document',
                hidden: ['documentId'],
                widgets: ['documentTypeId', 'issueDate', 'expiryDate', 'check']
            }]
        }
    },
    layouts: {
        edit: [
            { id: 'permission', label: 'Permission settings', widgets: ['roleCategory', 'permission', 'permissionRole'] },
            { id: 'document', label: 'Documents', widgets: ['document'] }
        ]
    }
};
