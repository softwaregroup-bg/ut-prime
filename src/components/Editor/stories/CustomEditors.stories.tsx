import React from 'react';
import Editor from '..';
export {default} from '../Editor.stories';

export const CustomEditors = () => {
    function Period({Input, Label, ErrorLabel}) {
        return (
            <>
                <ErrorLabel/>
                <div className='field grid w-full mx-0'>
                    <Label className='md:col-2' name='period' />
                    <Input fieldClass='md:col-4' name='period' />
                    <Input fieldClass='md:col-6' name='unit' />
                </div>
            </>
        );
    }

    Period.properties = ['period', 'unit'];

    return (<Editor
        id={1}
        onGet={() => Promise.resolve({
            period: 5,
            unit: 'days'
        })}
        onDropdown={names => Promise.resolve({})}
        schema={{
            properties: {
                unit: {
                    widget: {
                        type: 'select',
                        options: [
                            {value: 'minutes', label: 'Minutes'},
                            {value: 'hours', label: 'Hours'},
                            {value: 'days', label: 'Days'},
                            {value: 'months', label: 'Months'}
                        ]
                    }
                },
                period: {
                    title: 'Expiration period',
                    type: 'integer'
                }
            }
        }}
        cards={{
            edit: {
                widgets: ['Period']
            }
        }}
        editors={{
            Period
        }}
    />);
};
