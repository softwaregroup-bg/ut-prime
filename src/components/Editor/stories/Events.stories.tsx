import React from 'react';
import type { Story } from '@storybook/react';
import type { Props } from '../Editor.types';
import Editor from '..';
import useToast from '../../hooks/useToast';

import {input, dropdowns} from '../../test/input';

export {default} from '../Editor.stories';

const Template: Story<Props> = ({methods, ...args}) => {
    const {toast, submit} = useToast();
    return <>
        <Editor
            id={1}
            object='input'
            onGet={() => Promise.resolve({input: {}})}
            onDropdown={() => Promise.resolve(dropdowns)}
            onAdd={submit}
            onEdit={submit}
            onFieldChange='handleFieldChange'
            methods={{
                async handleFieldChange({field, value, event}: {field: unknown, value, event: Event}) {
                    submit({field, value});
                    // throw new Error('test error');
                    // return false;
                },
                async handleAutocomplete() {
                    return {
                        suggestions: [{
                            value: 1, label: 'value 1', status: 'active'
                        }, {
                            value: 2, label: 'value 2', status: 'pending'
                        }, {
                            value: 3, label: 'value 3', status: 'new'
                        }]
                    };
                },
                async 'portal.customization.get'() {
                    return {};
                },
                ...methods
            }}
            {...args}
        />
        {toast}
    </>;
};

declare type StoryTemplate = Story<Partial<Props>> & {
    play: (context: {canvasElement: HTMLElement}) => Promise<void>
}

export const Events: StoryTemplate = Template.bind({});
Events.args = {
    ...input,
    layouts: {edit: ['left', 'center', 'right']}
};

export const EventsFormAPI: StoryTemplate = Template.bind({});
EventsFormAPI.args = {
    schema: {
        properties: {
            a: {
                type: 'integer',
                widget: {onChange: 'handleA'}
            },
            b: {
                type: 'integer',
                widget: {onChange: 'handleB'}
            },
            sum: {
                type: 'number',
                readOnly: true
            },
            discount: {
                type: 'number',
                widget: {
                    onChange: 'handleDiscount',
                    enabled: '$.calc.positive'
                }
            },
            bitmask: {
                widget: {onChange: 'handleBitmask'}
            },
            input: {
                widget: {
                    visible: 'visible',
                    enabled: 'enabled'
                }
            },
            visible: {
                type: 'boolean'
            },
            enabled: {
                type: 'boolean'
            }
        }
    },
    cards: {
        edit: {
            className: 'lg:col-3',
            widgets: [
                'a',
                'b',
                'sum',
                'discount',
                'bitmask',
                'visible',
                'enabled',
                'input'
            ]
        }
    },
    onGet: () => Promise.resolve({a: 0, b: 0, sum: 0, visible: true, enabled: true}),
    methods: {
        async handleA({form, value}) {
            const sum = Number(form.getValues('b')) + Number(value);
            form.setValue('sum', sum);
            form.setValue('$.calc.positive', sum > 0);
        },
        async handleB({form, value}) {
            const sum = Number(form.getValues('a')) + Number(value);
            form.setValue('sum', sum);
            form.setValue('$.calc.positive', sum > 0);
        },
        async handleDiscount({value, form}) {
            if (Number(value) < 0) throw new Error('Discount must be > 0'); // throwing means abort the edit and show error
            if (Number(value) >= 0) form.clearErrors('discount');
        },
        async handleBitmask({value}) {
            if (/[^01]/.test(value)) return false; // returning false means abort the edit silently
        }
    }
};
