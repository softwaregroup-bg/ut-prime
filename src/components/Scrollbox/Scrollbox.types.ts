import React from 'react';
import {createUseStyles} from 'react-jss';

export interface Props {
    className?: string;
    children: any;
}

export type ComponentProps = React.FC<Props>

export const useStyles = createUseStyles({
    scrollbox: {
    }
})