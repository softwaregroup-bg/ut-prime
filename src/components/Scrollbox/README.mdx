# Scrollbox
import { Canvas, Story, ArgsTable } from "@storybook/addon-docs";
import {Input, Table} from "./Scrollbox.stories";

## Importing

```jsx
import Scrollbox from 'ut-prime/core/Scrollbox';

<Scrollbox className='name' />;
```

## Basic Usage

<Canvas>
    <Story id="scrollbox--basic" />
</Canvas>

## Properties

<ArgsTable story="."/>
