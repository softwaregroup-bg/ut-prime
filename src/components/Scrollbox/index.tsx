import React from 'react';
import clsx from 'clsx';
import type {ComponentProps} from './Scrollbox.types';
import {useStyles} from './Scrollbox.types';
import useScroll from '../hooks/useScroll';

const Scrollbox: ComponentProps = ({
    noScroll,
    className,
    ...rest
}) => {
    const styles = useStyles();
    const [ref, style] = useScroll(noScroll);
    return <div
        className={clsx('overflow-y-auto', styles.scrollbox, className)}
        {...rest}
        {...!noScroll && {ref, style}}
    />;
};

export default Scrollbox;
