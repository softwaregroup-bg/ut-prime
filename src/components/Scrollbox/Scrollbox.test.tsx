import React from 'react';

import { render } from '../test';
import { Basic } from './Scrollbox.stories';

describe('<Scrollbox />', () => {
    it('Basic render equals snapshot', async() => {
        const { findByTestId } = render(<Basic {...Basic.args} />);
        expect(await findByTestId('ut-front-test')).toMatchSnapshot();
    });
});
