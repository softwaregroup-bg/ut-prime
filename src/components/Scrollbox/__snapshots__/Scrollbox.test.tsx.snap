// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Scrollbox /> Basic render equals snapshot 1`] = `
<div
  data-testid="ut-front-test"
>
  <div
    class="overflow-y-auto scrollbox-0-2-1 col-2"
    style="max-height: 768px;"
  >
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Laboriosam totam expedita adipisci perspiciatis unde magni et ipsa quis possimus quam? Commodi eligendi quisquam voluptatem consequatur, iste ut sunt vel expedita autem omnis blanditiis ex id, asperiores fugit ducimus obcaecati. Maxime excepturi ex ipsa id praesentium! Molestias consequatur eos, amet nemo aspernatur necessitatibus optio. Nisi, assumenda odio exercitationem quia iure ducimus eos non? Asperiores quasi labore, iste, inventore molestias natus, eaque error praesentium ipsa placeat perspiciatis in expedita illo similique. Odio quis maxime exercitationem fugit, possimus vero rem vitae neque similique, modi omnis quidem recusandae nostrum corporis voluptatem harum culpa eaque sit veniam? Saepe nobis sed excepturi, in repellendus expedita cumque adipisci temporibus deleniti ex quis? Nemo doloribus corrupti impedit? Officiis odio veritatis animi aspernatur voluptatum, repudiandae magni doloremque aut commodi labore reiciendis eaque corporis deserunt nam, expedita dolores, ab et? Reprehenderit dolore suscipit debitis provident perferendis, minima cumque tenetur voluptatibus magnam beatae earum! Fuga adipisci maxime distinctio aspernatur sit unde, impedit, nulla quibusdam cupiditate blanditiis ut, provident veniam dolore aperiam quidem perspiciatis. Eligendi illum itaque dicta perferendis temporibus, molestias rerum excepturi nesciunt qui repellat cum voluptatem iusto quasi magnam libero eius labore maxime nemo? Ex nobis blanditiis incidunt eos. Autem architecto dignissimos illo cumque rem ipsum ratione quam dolorem quae perspiciatis possimus praesentium, tempore eius quo eligendi placeat odio doloremque mollitia repellendus vitae optio quos cupiditate quisquam accusantium! Id autem cumque a saepe consequatur! Hic modi vero, officiis, sit eum corporis error fugiat et non maxime laudantium illo voluptatum eveniet, dolor excepturi assumenda unde necessitatibus iure labore itaque totam vitae delectus consequuntur ab. Quod distinctio odio est esse cumque officiis cupiditate aperiam magnam rerum, hic sed incidunt temporibus enim beatae veritatis iure non aspernatur voluptates ab quo, aut quasi neque minus?
    </p>
  </div>
</div>
`;
